usersync/encoder.go

```diff
@@ -0,0 +1,19 @@
+package usersync
+
+import (
+	"encoding/base64"
+	"encoding/json"
+)
+
+type Encoder interface {
+	Encode(c *Cookie) string // Encode a cookie into a base 64 string
+}
+
+type EncoderV1 struct{}
+
+func (e EncoderV1) Encode(c *Cookie) string {
+	j, _ := json.Marshal(c)
```

Comment:
This can fail due to memory corruption and I believe if `c` is nil. I recommend you change the return types to `(string, error)` and bubble the marshal error up. Please also add a test for when `c` is nil and if it fails, consider returning an empty string as a special case.

Reply:
Note for me to bring up the best way to handle this error bubble up throughout the refactor at the next team time.